packages/api/src/webhooks/cannon.ts

```diff
@@ -0,0 +1,204 @@
+      }
+    } catch (e) {
+      console.error("error: ", e);
+      throw e;
+    }
+
+    // This is mainly useful for local testing
+    if (user.admin) {
+      isLocal = false;
+    } else {
+      try {
+        if (ips && ips.length) {
+          isLocal = isLocalIP(ips[0]);
+        } else {
+          isLocal = true;
+        }
+      } catch (e) {
+        console.error("isLocal Error", isLocal, e);
+        throw e;
+      }
+    }
+    if (isLocal) {
+      // don't fire this webhook.
+      console.log(
+        `webhook ${webhook.id} resolved to a localIP, url: ${webhook.url}, resolved IP: ${ips}`
+      );
+    } else {
+      console.log("preparing to fire webhook ", webhook.url);
+      // go ahead
+      let params = {
+        method: "POST",
+        headers: {
+          "content-type": "application/json",
+          "user-agent": "livepeer.com",
+        },
+        timeout: WEBHOOK_TIMEOUT,
+        body: JSON.stringify({
+          id: webhook.id,
+          event: webhook.event,
+          stream: sanitized,
+        }),
+      };
+
+      try {
+        logger.info(`webhook ${webhook.id} firing`);
+        let resp = await fetchWithTimeoutAndSleep(
+          webhook.url,
+          params,
+          event.lastInterval
+        );
+        if (resp.status >= 200 && resp.status < 300) {
+          // 2xx requests are cool.
+          // all is good
+          logger.info(`webhook ${webhook.id} fired successfully`);
+          await this.storeResponse(webhook, event, resp);
+          return true;
+        }
+        console.error(
+          `webhook ${webhook.id} didn't get 200 back! response status: ${resp.status}`
+        );
+        await this.storeResponse(webhook, event, resp);
+        // retry
+        this.retry(event);
+        return;
+        // return !webhook.blocking;
+      } catch (e) {
+        console.log("firing error", e);
+        // return !webhook.blocking;
+        return;
+      }
+    }
+  }
+
+  async storeResponse(
+    webhook: Webhook,
+    event: Queue,
+    resp: Response,
+    duration = 0
+  ) {
+    // TODO
+    // store response for each time a webhook fires.
+    await this.db.webhookResponse.create({
+      webhookId: webhook.id,
+      eventId: event.id,
+      statusCode: resp.status,
+      response: resp,
+    });
+  }
+
+  async onTrigger(event: Queue) {
+    if (!event) {
+      // throw new Error('onTrigger requires a Queue event!')
+      // this is fine because pop could return a null if some other client raced and picked the task
+      // faster when it got the NOTIFY call.
+      return;
+    }
+
+    const { data: webhooksList } = await getWebhooks(
+      this.store,
+      event.userId,
+      event.event
+    );
+
+    let stream = await this.db.stream.get(event.streamId);
+    // basic sanitization.
+    let sanitized = { ...stream };
+    delete sanitized.streamKey;
+
+    let user = await this.db.user.get(`user/${event.userId}`);
```

Comment:
```suggestion
    let user = await this.db.user.get(event.userId);
```
Also, what would happen if user is not found?